lte/gateway/c/oai/tasks/sgw_s8/sgw_s8_handlers.c

```diff
@@ -561,3 +578,267 @@ static int sgw_s8_add_gtp_tunnel(
   }
   OAILOG_FUNC_RETURN(LOG_SGW_S8, rv);
 }
+
+void sgw_s8_handle_modify_bearer_request(
+    sgw_state_t* state,
+    const itti_s11_modify_bearer_request_t* const modify_bearer_pP,
+    imsi64_t imsi64) {
+  OAILOG_FUNC_IN(LOG_SGW_S8);
+
+  uint8_t idx                                                    = 0;
+  uint8_t sgi_rsp_idx                                            = 0;
+  itti_sgi_update_end_point_response_t sgi_update_end_point_resp = {0};
+  struct in_addr enb                  = {.s_addr = 0};
+  struct in_addr pgw                  = {.s_addr = 0};
+  sgw_eps_bearer_ctxt_t* bearer_ctx_p = NULL;
+
+  OAILOG_INFO_UE(
+      LOG_SGW_S8, imsi64, "Rx MODIFY_BEARER_REQUEST, teid " TEID_FMT "\n",
+      modify_bearer_pP->teid);
+
+  sgw_eps_bearer_context_information_t* sgw_context_p =
+      sgw_get_sgw_eps_bearer_context(modify_bearer_pP->teid);
+  if (!sgw_context_p) {
+    OAILOG_ERROR_UE(
+        LOG_SGW_S8, imsi64,
+        "Failed to fetch sgw_eps_bearer_context_info from "
+        "context_teid " TEID_FMT " \n",
+        modify_bearer_pP->teid);
+    if ((send_mbr_failure(LOG_SGW_S8, modify_bearer_pP, imsi64) != RETURNok)) {
+      OAILOG_ERROR(
+          LOG_SGW_S8,
+          "Error in sending modify bearer response to MME App for context "
+          "teid " TEID_FMT "\n",
+          modify_bearer_pP->teid);
+    }
+    OAILOG_FUNC_OUT(LOG_SGW_S8);
+  }
+  sgw_context_p->trxn                    = modify_bearer_pP->trxn;
+  sgi_update_end_point_resp.context_teid = modify_bearer_pP->teid;
+  // Traversing through the list of bearers to be modified
+  for (; idx <
```

Comment:
let's define the `idx` here on the context where it's gonna be used instead of the top of the function